ext/winevt/winevt_utils.cpp

```diff
@@ -596,8 +596,57 @@ get_description(EVT_HANDLE handle, LANGID langID, EVT_HANDLE hRemote)
   return _wcsdup(result.data());
 }
 
+static int ExpandSIDWString(PSID sid, CHAR **out_expanded)
+{
+#define MAX_NAME 256
+  DWORD len = MAX_NAME, err = ERROR_SUCCESS;
+  SID_NAME_USE sid_type = SidTypeUnknown;
+  char account[MAX_NAME];
+  char domain[MAX_NAME];
+  DWORD result_len = 0;
+  CHAR *formatted = NULL;
+  VALUE vformatted;
+
+  if (!LookupAccountSidA(NULL, sid,
+                         account, &len, domain,
+                         &len, &sid_type)) {
+    err = GetLastError();
+    if (err == ERROR_NONE_MAPPED) {
+      goto not_mapped_error;
+    }
+    else {
+      return -2;
+    }
+
+    goto error;
+  }
+
+  result_len = strlen(domain) + 1 + strlen(account) + 1;
+  formatted = (CHAR *)ALLOCV(vformatted, result_len);
+  if (formatted == NULL) {
+    goto error;
+  }
+
+  _snprintf_s(formatted, result_len, _TRUNCATE, "%s\\%s", domain, account);
+
+  *out_expanded = strdup(formatted);
```

Comment:
Sorry, I'm not familiar with this, but I wonder who manages the memory of the result of this `strdup()`.
Is this safe?

Reply:
This should be safe. We shouldn't returned back allocated memory with ALLOCV.
Instead, we need to allocate heap memory to return the result.

Reply:
Shouldn't it be released after this?
https://github.com/fluent-plugins-nursery/winevt_c/blob/360e2b776a3b1f5824c2cf0770e3a56edecf3030/ext/winevt/winevt_utils.cpp#L843

Reply:
If this is not safe, it might be a good idea to create and return a Ruby object from the beginning, such as using `rb_sprintf()`.

Reply:
https://github.com/fluent-plugins-nursery/winevt_c/pull/46/commits/9987636117b0e0af6ddbbeb10d820c7a74c28f18